### DISCUSSION TITLE
Nominatim or OSRM? Get Hotels around a coordinate

### DISCUSSION THREAD

**Discussion by ppapasav** [osm-search/Nominatim#2457] [Q&A] [answered]
I'm trying to figure out a way to extract hotel or some other point of interest given either

1. Radius around a coordinate
2. Two sets of coordinates forming a box

I can successfully get a reverse lookup using something like

https://nominatim.openstreetmap.org/reverse?format=geojson&lat=51.5070&lon=-.152195

But, if the GPS is off even a little, I'll have the wrong location. So, given the scenario that a user is trying to select the location they're at, I'll know the phone's precision at that time - It may be 200 meters, as an example. How could I structure a reverse lookup using the given coordinates of the user's device and do a radius for the precision ... Or I could take that precision and derive a box around that as well - Will Nominatim return a list of locations around there?

Also, is there a way to specify the types of locations we want to return either hotels or some other point of interest?

Thank you

**Comment by mtmail** [comment, marked as answer]
Nominatim reverse search will only return one result, never several.

Have a look at a forward search using special phrases. https://wiki.openstreetmap.org/wiki/Nominatim/Special_Phrases  Forward search also works with coordinates.

"hotel near 51.5070,-0.152195" https://nominatim.openstreetmap.org/ui/search.html?q=hotel+near+51.5070%2C-0.152195

As https://nominatim.org/release-docs/latest/api/Search/ explains it's better to use Overpass API (https://overpass-api.de/) for these kinds of queries. Nominatim has only a fixed set of phrases that work (hotel is one of the phrases), they're not updated and not very flexible.

**Comment by ppapasav** [comment]
Thanks for this - Good info!
